Frends.RabbitMQ.Read/Frends.RabbitMQ.Read.Tests/UnitTests.cs

```diff
@@ -1,263 +1,621 @@
-                var newUri = new Uri(connection.Host);
-                return (connectionHelper.AMQPConnection.Endpoint.HostName != newUri.Host);
-            case AuthenticationMethod.Host:
-                return (connectionHelper.AMQPConnection.Endpoint.HostName != connection.Host);
-            default:
-                throw new ArgumentException($"IsConnectionHostNameChanged: AuthenticationMethod missing.");
-        }
-    }
+using Frends.RabbitMQ.Read.Definitions;
+using Frends.RabbitMQ.Read.Tests.Lib;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using RabbitMQ.Client;
+using System.Text;
+
+namespace Frends.RabbitMQ.Read.Tests;
+
+[TestClass]
+public class UnitTests
+{
+    /// <summary>
+    /// You will need access to RabbitMQ queue, you can create it e.g. by running
+    /// docker run -d --hostname my-rabbit -p 5672:5672 -p 8080:1567 -p 15672:15672 -e RABBITMQ_DEFAULT_USER=agent -e RABBITMQ_DEFAULT_PASS=agent123  rabbitmq:3.9-management
+    /// In that case URI would be amqp://agent:agent123@localhost:5672 
+    /// Access UI from http://localhost:15672 username: agent, password: agent123
+    /// </summary>
+
+    private const string TestUri = "amqp://agent:agent123@localhost:5672";
+    private const string TestHost = "localhost";
+
+    [TestInitialize]
+    public void CreateExchangeAndQueue()
+    {
+        var factory = new ConnectionFactory { Uri = new Uri(TestUri) };
+        using var connection = factory.CreateConnection();
+        using var channel = connection.CreateModel();
+        channel.ExchangeDeclare("exchange", type: "fanout", durable: false, autoDelete: false);
+        channel.QueueDeclare("queue", durable: false, exclusive: false, autoDelete: false);
+        channel.QueueBind("queue", "exchange", routingKey: "");
+    }
+
+    [TestCleanup]
+    public void DeleteExchangeAndQueue()
+    {
+        var factory = new ConnectionFactory { Uri = new Uri(TestUri) };
+        using var connection = factory.CreateConnection();
+        using var channel = connection.CreateModel();
+        channel.QueueDelete("queue", false, false);
+        channel.ExchangeDelete("exchange", ifUnused: false);
+    }
+
+    /// <summary>
+    /// Connect with hostname and read single message.
+    /// </summary>
+    [TestMethod]
+    public void TestReadSingleMessageWithHostAutoNack()
+    {
+        Connection connection = new()
+        {
+            Host = TestHost,
+            Username = "agent",
+            Password = "agent123",
+            RoutingKey = "queue",
+            QueueName = "queue",
+            AuthenticationMethod = AuthenticationMethod.Host,
+            ExchangeName = null,
+
+            AutoAck = ReadAckType.AutoAck,
+            ReadMessageCount = 1,
+        };
+
+        Publish(connection, 1);
+        var result = RabbitMQ.Read(connection);
+
+        Assert.IsTrue(result.MessagesBase64.Count == 1 && result.MessageUTF8.Count == 1 && result.Success.Equals(true));

```

Comment:
This check should be split into their own asserts and use `AreEqual` for equal comparisons.